pkg/usage/metrics.go

```diff
@@ -0,0 +1,208 @@
+	"k8s.io/klog"
+)
+
+type Metrics struct {
+	pageViewsTotal       *prometheus.CounterVec
+	pageImpressionsTotal *prometheus.CounterVec
+	users                *prometheus.GaugeVec
+}
+
+func (m *Metrics) GetCollectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		m.pageViewsTotal,
+		m.pageImpressionsTotal,
+		m.users,
+	}
+}
+
+func (m *Metrics) HandlePageView(perspective string) error {
+	counter, err := m.pageViewsTotal.GetMetricWithLabelValues(perspective)
+	if counter != nil && err == nil {
+		counter.Inc()
+	}
+	return err
+}
+
+func (m *Metrics) HandlePageImpression(perspective string) error {
+	counter, err := m.pageImpressionsTotal.GetMetricWithLabelValues(perspective)
+	if counter != nil && err == nil {
+		counter.Inc()
+	}
+	return err
+}
+
+func (m *Metrics) MonitorUsers(
+	userSettingsClient *http.Client,
+	userSettingsEndpoint string,
+	serviceAccountToken string,
+) {
+	go func() {
+		time.Sleep(3 * time.Second)
+		go m.updateUsersMetric(userSettingsClient, userSettingsEndpoint, serviceAccountToken)
+	}()
+
+	ticker := time.NewTicker(1 * time.Minute) // TODO: Change to every 12 hours?
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				m.updateUsersMetric(userSettingsClient, userSettingsEndpoint, serviceAccountToken)
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (m *Metrics) updateUsersMetric(
+	userSettingsClient *http.Client,
+	userSettingsEndpoint string,
+	serviceAccountToken string,
+) error {
+	klog.Info("usage.Metrics: Count console users...\n")
+	startTime := time.Now()
+
+	ctx := context.TODO()
+	config := &rest.Config{
+		Host:        userSettingsEndpoint,
+		BearerToken: serviceAccountToken,
+		Transport:   userSettingsClient.Transport,
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("usage.Metrics: Failed to create service account client: %v\n", err)
+		return err
+	}
+
+	roleBindingList, err := client.RbacV1().RoleBindings("openshift-console-user-settings").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("usage.Metrics: Failed to get user-settings RoleBindings: %v\n", err)
+		return err
+	}
+
+	canGetNamespaceAccessReview := &authv1.SelfSubjectAccessReview{
+		Spec: authv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authv1.ResourceAttributes{
+				Verb:     "get",
+				Resource: "namespaces",
+			},
+		},
+	}
+
+	var clusterAdmins, developers, unknowns float64
+
+	for _, roleBinding := range roleBindingList.Items {
+		if !strings.HasPrefix(roleBinding.Name, "user-settings-") || !strings.HasSuffix(roleBinding.Name, "-rolebinding") {
+			klog.Infof("usage.Metrics: Ignore role binding: %s (name doesn't match user-settings-*-rolebinding)\n", roleBinding.Name)
+			continue
+		}
+		if len(roleBinding.Subjects) != 1 {
+			klog.Infof("usage.Metrics: Ignore role binding: %s (unexpected subject length %v != 1)\n", roleBinding.Name, len(roleBinding.Subjects))
```

Comment:
The %v is mainly used for complex variables like arrays, maps, errors , .... names should be quoted and numbers should be also formated accordingly
```suggestion
			klog.Infof("usage.Metrics: Ignore role binding: %q (unexpected subject length %i != 1)\n", roleBinding.Name, len(roleBinding.Subjects))
```